### PR TITLE
Add OpenGL Shim

### PR DESCRIPTION
**Pull request by Etaash-mathamsetty**

### CODE REVIEW

File: src/gl/shim.c [hunk 1]
```diff
@@ -36,7 +48,7 @@ static void loadMangoHud() {
 
     if (mangoHudLoaded) {
         printf("shim: Loaded mangohud library\n");
```
Comment (flightlessmango) [inline src/gl/shim.c:50]: We don't want to print to stdout from shim, use stderr

File: src/meson.build [hunk 2]
```diff
@@ -222,6 +222,7 @@ mangohud_opengl_shared_lib = shared_library(
   opengl_files,
   vklayer_files,
   util_files,
+  gnu_symbol_visibility : 'hidden',
```
Comment (flightlessmango) [inline src/meson.build:225]: Not sure if you or I added this, but it seems irrelevant to this PR?

Comment (Etaash-mathamsetty) [reply src/meson.build:225]: it was part of your original code, I thought it was maybe used with the DEEPBIND for something
